### PR TITLE
Don't crash in initializer

### PR DESCRIPTION
**Pull request by biafra23**
Handles #1398

Changes proposed in this pull request:
- Capture exception in initializing unstoppable dependency
- This does not fix the bug itself but it prevents crashing
- Until the lib gets an update, we should probably remove the menu entry.

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt [hunk 1]
```diff
@@ -51,20 +53,86 @@ class RepositoryModule {
...
-        val network = Network.valueOf(firstWord.toUpperCase());
-        return Resolution.builder()
+        val firstWord = BuildConfig.BLOCKCHAIN_NET_URL.removePrefix("https://").split(".").first()
+        val network = Network.valueOf(firstWord.toUpperCase())
+        return try {
+            Resolution.builder()
                 .chainId(NamingServiceType.CNS, network)
                 .infura(NamingServiceType.CNS, BuildConfig.INFURA_API_KEY)
                 .build()
+        } catch (throwable: Throwable) {
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt:63]: Is there a better way than providing an "empty" interface? If not, then I'd move this to separate file called DummyDomainResolution.

File: app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt [hunk 2]
```diff
@@ -51,20 +53,86 @@ class RepositoryModule {
...
+            Timber.e(throwable, "Error initializing UnstoppableDomains")
+            object : DomainResolution {
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt:65]: Is there a better way than providing an "empty" interface? If not, then I'd move this to separate file called DummyDomainResolution.

File: app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt [hunk 3]
```diff
@@ -51,20 +50,25 @@ class RepositoryModule {
...
+        val firstWord = BuildConfig.BLOCKCHAIN_NET_URL.removePrefix("https://").split(".").first()
+        val network = Network.valueOf(firstWord.toUpperCase())
+        return try {
+            Resolution.builder()
                 .chainId(NamingServiceType.CNS, network)
                 .infura(NamingServiceType.CNS, BuildConfig.INFURA_API_KEY)
                 .build()
+        } catch (throwable: Throwable) {
+            Timber.e(throwable, "Error initializing UnstoppableDomains")
+            DummyDomainResolution()
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt:62]: wouldn't it lead to a crash when using this dummy?

**Comment by elgatovital** [comment]
tests are failing

File: app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt (hunk 3 above)
Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt:62]: No. It leads to a generic error :-)

Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt:62]: image

**Comment by biafra23** [comment]
I don't get what the failing tests have to do with this PR. This is the message I found in the logs:

com.android.ddmlib.InstallException: Unknown failure: Error: Could not access the Package Manager.  Is the system running?

**Review by elgatovital — APPROVED** [review]

**Pull request merged**